website/src/components/ToSWrapper.tsx

```diff
@@ -0,0 +1,49 @@
+import { Flex, Text } from "@chakra-ui/react";
+import { useSession } from "next-auth/react";
+import { useTranslation } from "next-i18next";
+import { ReactNode, useMemo } from "react";
+import { SubmitButton } from "src/components/Buttons/Submit";
+import { SurveyCard } from "src/components/Survey/SurveyCard";
+import { post } from "src/lib/api";
+import { TermsOfService } from "src/pages/terms-of-service";
+
+const navigateAway = () => {
+  location.href = "https://laion.ai/";
+};
+
+const acceptToS = async () => {
+  await post("/api/tos", { arg: {} });
+  location.reload();
+};
+
+export const ToSWrapper = ({ children }: { children?: ReactNode | undefined }) => {
+  const { t } = useTranslation("tos");
+  const { data: session } = useSession();
+  const hasAcceptedTos = Boolean(session?.user.tosAcceptanceDate);
```

Comment:
we should check for status === "loading" here. There is flash when hard reloading the page now because session is null

Reply:
https://github.com/LAION-AI/Open-Assistant/pull/1080/commits/61914e2f26d02e43d80230f2d5888d38b1d58c4e